turbopy/diagnostics.py

```diff
@@ -95,12 +98,15 @@ def __init__(self, owner: Simulation, input_data: dict):
         self.field = None
         self.output_function = None
         self.csv = None
+        self.interval = None
 
     def diagnose(self):
         """
         Run output function given the value of the field.
         """
         self.output_function(self.get_value(self.field))
+        if self._owner.clock.time % self.interval == 0:
+            self.csv.finalize()
```

Comment:
So, I admit that my code for this pretty messy, and this would be easy to miss, but the way you have this written introduces a possible bug. If the user did not select `"csv"` output, then I don't know what would happen on line 109... probably the code would crash.

I have an idea for how to fix this. But it involves several changes working together, so I'll describe the idea in the main review comment.